plugin-server/src/worker/ingestion/event-pipeline/runAsyncHandlersStep.ts

```diff
@@ -10,25 +10,40 @@ export async function runAsyncHandlersStep(
     person: Person | undefined,
     elements: Element[] | undefined
 ): Promise<StepResult> {
+    await Promise.all([processOnEvent(runner, event), processOnActionAndWebhooks(runner, event, person, elements)])
+
+    return null
+}
+
+async function processOnEvent(runner: EventPipelineRunner, event: PreIngestionEvent) {
+    const processedPluginEvent = convertToProcessedPluginEvent(event)
+    const isSnapshot = event.event === '$snapshot'
+    const method = isSnapshot ? runOnSnapshot : runOnEvent
+
+    await runInstrumentedFunction({
+        server: runner.hub,
+        event: processedPluginEvent,
+        func: (event) => method(runner.hub, event),
+        statsKey: `kafka_queue.single_${isSnapshot ? 'on_snapshot' : 'on_event'}`,
+        timeoutMessage: `After 30 seconds still running ${isSnapshot ? 'onSnapshot' : 'onEvent'}`,
+    })
+}
+
+async function processOnActionAndWebhooks(
+    runner: EventPipelineRunner,
+    event: PreIngestionEvent,
+    person: Person | undefined,
+    elements: Element[] | undefined
+) {
     const promises = []
     let actionMatches: Action[] = []
+    const processedPluginEvent = convertToProcessedPluginEvent(event)
+
     if (event.event !== '$snapshot') {
         actionMatches = await runner.hub.actionMatcher.match(event, person, elements)
-        promises.push(runner.hub.hookCannon.findAndFireHooks(event, person, event.siteUrl, actionMatches))
+        promises.push(runner.hub.hookCannon.findAndFireHooks(event, person, actionMatches))
```

Comment:
note that we're no longer awaiting these anywhere - we don't return them so `runAsyncHandlers` step doesn't await them. This function will just fire these promises and forget about them.

Reply:
This is a bug, good catch. There should be a Promise.all down here :)

Will fix.